<a href="https://colab.research.google.com/github/SuhelAnsari9272/Retails-Sales/blob/main/Retail_Sales_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns
import json

In [ ]:
#reading the dataset
#For Behaviour data
path_behv = '/content/drive/MyDrive/Retail_Sales_ABADS/behaviour.json'

with open(path_behv, 'r') as file:
    data_behv = json.load(file)

df_behv = pd.DataFrame()

for entry in data_behv:
    for key, value in entry.items():
        row_data = {"ID": key, **value}
        df_behv = df_behv.append(row_data, ignore_index=True)

#For Campaign data
path_camp = '/content/drive/MyDrive/Retail_Sales_ABADS/campaign.json'
with open(path_camp, 'r') as file:
    data_camp = json.load(file)

df_camp = pd.DataFrame()

for entry in data_camp:
    for key, value in entry.items():
        row_data = {"ID": key, **value}
        df_camp = df_camp.append(row_data, ignore_index=True)

#For Demographics data
path_demo = "/content/drive/MyDrive/Retail_Sales_ABADS/demographics.txt"
df_demo = pd.read_csv(path_demo, sep='\t')



In [ ]:
df_behv.head()

ID  Recency  MntWines  MntFruits  MntMeatProducts  MntFishProducts  \
0   ID_1826        0       189        104              379              111   
1      ID_1        0       464          5               64                7   
2  ID_10476        0       134         11               59               15   
3   ID_1386        0        10          0                1                0   
4   ID_5371        0         6         16               24               11   

   MntSweetProducts  MntGoldProds  NumDealsPurchases  NumWebPurchases  \
0               189           218                  1                4   
1                 0            37                  1                7   
2                 2            30                  1                3   
3                 0             0                  1                1   
4                 0            34                  2                3   

   NumCatalogPurchases  NumStorePurchases  NumWebVisitsMonth  
0                    4                  6                  1  
1                    3                  7                  5  
2                    2                  5                  2  
3                    0                  2                  7  
4                    1                  2                  7

In [ ]:
df_camp.head()


ID  AcceptedCmp1  AcceptedCmp2  AcceptedCmp3  AcceptedCmp4  \
0   ID_1826             0             0             0             0   
1      ID_1             0             1             0             0   
2  ID_10476             0             0             0             0   
3   ID_1386             0             0             0             0   
4   ID_5371             0             0             1             0   

   AcceptedCmp5  Response  Complain  
0             0         1         0  
1             0         1         0  
2             0         0         0  
3             0         0         0  
4             0         1         0

In [ ]:
df_demo.head()

ID  Year_Birth   Education Marital_Status      Income   Kidhome  \
0   1826        1970  Graduation       Divorced  $84,835.00         0   
1      1        1961  Graduation         Single  $57,091.00         0   
2  10476        1958  Graduation        Married  $67,267.00         0   
3   1386        1967  Graduation       Together  $32,474.00         1   
4   5371        1989  Graduation         Single  $21,474.00         1   

   Teenhome Dt_Customer Country  
0         0     6/16/14      SP  
1         0     6/15/14      CA  
2         1     5/13/14      US  
3         1     5/11/14     AUS  
4         0      4/8/14      SP

In [ ]:
#Data Description
df_behv.describe()

Recency     MntWines    MntFruits  MntMeatProducts  \
count  2240.000000  2240.000000  2240.000000      2240.000000   
mean     49.109375   303.935714    26.302232       166.950000   
std      28.962453   336.597393    39.773434       225.715373   
min       0.000000     0.000000     0.000000         0.000000   
25%      24.000000    23.750000     1.000000        16.000000   
50%      49.000000   173.500000     8.000000        67.000000   
75%      74.000000   504.250000    33.000000       232.000000   
max      99.000000  1493.000000   199.000000      1725.000000   

       MntFishProducts  MntSweetProducts  MntGoldProds  NumDealsPurchases  \
count      2240.000000       2240.000000   2240.000000        2240.000000   
mean         37.525446         27.062946     44.021875           2.325000   
std          54.628979         41.280498     52.167439           1.932238   
min           0.000000          0.000000      0.000000           0.000000   
25%           3.000000          1.000000      9.000000           1.000000   
50%          12.000000          8.000000     24.000000           2.000000   
75%          50.000000         33.000000     56.000000           3.000000   
max         259.000000        263.000000    362.000000          15.000000   

       NumWebPurchases  NumCatalogPurchases  NumStorePurchases  \
count      2240.000000          2240.000000        2240.000000   
mean          4.084821             2.662054           5.790179   
std           2.778714             2.923101           3.250958   
min           0.000000             0.000000           0.000000   
25%           2.000000             0.000000           3.000000   
50%           4.000000             2.000000           5.000000   
75%           6.000000             4.000000           8.000000   
max          27.000000            28.000000          13.000000   

       NumWebVisitsMonth  
count        2240.000000  
mean            5.316518  
std             2.426645  
min             0.000000  
25%             3.000000  
50%             6.000000  
75%             7.000000  
max            20.000000

* On an average, the customers returns after <b> 50 days </b> .
* The average amount of expense in  <b> wines </b> is approx <b> 304 </b>.
* The average amount of expense in <b> fruits </b> is approx <b> 26 </b>.
* The average amount of expense in <b> meats </b>  is approx <b> 167 </b>.
* The average amount of expense in <b> fish </b> is approx <b> 38 </b>.
* The average amount of expense in <b> Sweets </b> is approx <b> 27 </b>.
* The average amount of expense in <b> Gold products </b> is approx <b> 44 </b>.
* On an average, <b> 2 </b> customers, make a purchase with a <b> discount </b>.
* On an average, <b> 4 </b> customers, make a purchase by <b> Web </b>.
* On an average, <b> 3 </b> customers, make a purchase by <b> catalog </b>.
* On an average, <b> 6 </b> customers, make a purchase by visting <b> store </b>.
* On an average, <b> 5 </b> customers, make a visits in <b> website </b>.


In [ ]:
df_camp.describe(include = 'all')

ID  AcceptedCmp1  AcceptedCmp2  AcceptedCmp3  AcceptedCmp4  \
count      2240   2240.000000   2240.000000   2240.000000   2240.000000   
unique     2240           NaN           NaN           NaN           NaN   
top     ID_1826           NaN           NaN           NaN           NaN   
freq          1           NaN           NaN           NaN           NaN   
mean        NaN      0.064286      0.013393      0.072768      0.074554   
std         NaN      0.245316      0.114976      0.259813      0.262728   
min         NaN      0.000000      0.000000      0.000000      0.000000   
25%         NaN      0.000000      0.000000      0.000000      0.000000   
50%         NaN      0.000000      0.000000      0.000000      0.000000   
75%         NaN      0.000000      0.000000      0.000000      0.000000   
max         NaN      1.000000      1.000000      1.000000      1.000000   

        AcceptedCmp5     Response     Complain  
count    2240.000000  2240.000000  2240.000000  
unique           NaN          NaN          NaN  
top              NaN          NaN          NaN  
freq             NaN          NaN          NaN  
mean        0.072768     0.149107     0.009375  
std         0.259813     0.356274     0.096391  
min         0.000000     0.000000     0.000000  
25%         0.000000     0.000000     0.000000  
50%         0.000000     0.000000     0.000000  
75%         0.000000     0.000000     0.000000  
max         1.000000     1.000000     1.000000

In [ ]:
df_demo.describe(include = 'all')

ID   Year_Birth   Education Marital_Status     Income   \
count    2240.000000  2240.000000        2240           2240        2216   
unique           NaN          NaN           5              8        1974   
top              NaN          NaN  Graduation        Married  $7,500.00    
freq             NaN          NaN        1127            864          12   
mean     5592.159821  1968.805804         NaN            NaN         NaN   
std      3246.662198    11.984069         NaN            NaN         NaN   
min         0.000000  1893.000000         NaN            NaN         NaN   
25%      2828.250000  1959.000000         NaN            NaN         NaN   
50%      5458.500000  1970.000000         NaN            NaN         NaN   
75%      8427.750000  1977.000000         NaN            NaN         NaN   
max     11191.000000  1996.000000         NaN            NaN         NaN   

            Kidhome     Teenhome Dt_Customer Country  
count   2240.000000  2240.000000        2240    2240  
unique          NaN          NaN         663       8  
top             NaN          NaN     8/31/12      SP  
freq            NaN          NaN          12    1095  
mean       0.444196     0.506250         NaN     NaN  
std        0.538398     0.544538         NaN     NaN  
min        0.000000     0.000000         NaN     NaN  
25%        0.000000     0.000000         NaN     NaN  
50%        0.000000     0.000000         NaN     NaN  
75%        1.000000     1.000000         NaN     NaN  
max        2.000000     2.000000         NaN     NaN

In [ ]:
#Data information
display(df_behv.info())
display(df_camp.info())
display(df_demo.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   ID                   2240 non-null   object
 1   Recency              2240 non-null   int64 
 2   MntWines             2240 non-null   int64 
 3   MntFruits            2240 non-null   int64 
 4   MntMeatProducts      2240 non-null   int64 
 5   MntFishProducts      2240 non-null   int64 
 6   MntSweetProducts     2240 non-null   int64 
 7   MntGoldProds         2240 non-null   int64 
 8   NumDealsPurchases    2240 non-null   int64 
 9   NumWebPurchases      2240 non-null   int64 
 10  NumCatalogPurchases  2240 non-null   int64 
 11  NumStorePurchases    2240 non-null   int64 
 12  NumWebVisitsMonth    2240 non-null   int64 
dtypes: int64(12), object(1)
memory usage: 227.6+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   ID            2240 non-null   object
 1   AcceptedCmp1  2240 non-null   int64 
 2   AcceptedCmp2  2240 non-null   int64 
 3   AcceptedCmp3  2240 non-null   int64 
 4   AcceptedCmp4  2240 non-null   int64 
 5   AcceptedCmp5  2240 non-null   int64 
 6   Response      2240 non-null   int64 
 7   Complain      2240 non-null   int64 
dtypes: int64(7), object(1)
memory usage: 140.1+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   ID              2240 non-null   int64 
 1   Year_Birth      2240 non-null   int64 
 2   Education       2240 non-null   object
 3   Marital_Status  2240 non-null   object
 4    Income         2216 non-null   object
 5   Kidhome         2240 non-null   int64 
 6   Teenhome        2240 non-null   int64 
 7   Dt_Customer     2240 non-null   object
 8   Country         2240 non-null   object
dtypes: int64(4), object(5)
memory usage: 157.6+ KB


None

* Income data type is needed to be changed to int from object